x-pack/plugins/cloud_security_posture/server/saved_objects/migrations/csp_rule_template.ts

```diff
@@ -43,9 +43,19 @@ function migrateCspRuleTemplatesToV870(
 ): SavedObjectUnsanitizedDoc<CspRuleTemplateV870> {
   // Keeps only metadata, deprecated state
   const { muted, enabled, ...attributes } = doc.attributes;
+
   return {
     ...doc,
-    attributes,
+    attributes: {
+      metadata: {
+        ...attributes.metadata,
+        benchmark: {
+          ...attributes.metadata.benchmark,
+          // CSPM introduced in 8.7, so we can assume all docs from 8.4.0 are KSPM
+          posture_type: 'kspm',
+        },
+      },
+    },
```

Comment:
this takes an `8.4.0` doc and returns an `8.7.0` doc, and since `CSPM` is introduced in `8.7.0`, we can say all `8.4.0` docs are `kspm`